### PR TITLE
#309 Delete an event as attendee

### PR DESCRIPTION
**Pull request by rezk2ll**
**demo**: https://streamable.com/ifqouf

### CODE REVIEW

File: src/esn.calendar.libs/app/components/event/form/event-form.controller.js [hunk 1]
```diff
@@ -298,11 +300,11 @@ function CalEventFormController(
     }
   }
 
-  function deleteEvent() {
+  function deleteEvent(allInstances = false) {
```
Comment (huy-ta) [inline src/esn.calendar.libs/app/components/event/form/event-form.controller.js:303]:
```suggestion
  function deleteEvent(shouldRemoveAllInstances = false) {
```

File: src/esn.calendar.libs/app/components/event/form/event-form.controller.js [hunk 2]
```diff
@@ -675,4 +677,28 @@ function CalEventFormController(
...
+          function deleteAllInstances() {
+            deleteEvent(true);
+          }
+        },
+        placement: 'center'
+      });
+    } else {
+      deleteEvent();
+    }
+  }
```
Comment (huy-ta) [inline src/esn.calendar.libs/app/components/event/form/event-form.controller.js:703]:
```suggestion
  function deteteEventForAttendee() {
    if (!$scope.editedEvent.isInstance()) return deleteEvent();
    
   $modal({
      template: require('./modals/delete-instance-or-series-modal.pug'),
      controller: /* @ngInject */ function($scope) {
        $scope.editChoice = 'this';

        $scope.submit = function() {
          $scope.$hide();

          ($scope.editChoice === 'this' ? deleteEvent : deleteAllInstances)();
        };

        function deleteAllInstances() {
          deleteEvent(true);
        }
      },
      placement: 'center'
    });
  }
```

**Review by huy-ta — COMMENTED** [review]
I'll take a closer look at this tomorrow.

**Comment by huy-ta** [comment]
Here's the list of bugs that I've found (I'll update them here once I've found new ones):

1. As an attendee, delete a recurrence exception and then accept the whole series:
    - Expected behaviour: The deleted recurrence exception should reappear and the organizer can see in the recurrence exception that the attendee has accepted.
    - Current behaviour: The deleted recurrence exception does not reappear although the organizer can still see in the recurrence exception that the attendee has accepted.

2. As an attendee, delete the whole series of a recurrent event:
    - Expected behaviour: An email should be sent to the organizer with its title beginning with "Declined..."
    - Current behaviour: An email is sent to the organizer with its title beginning with "Participation updated..."

**Review by huy-ta — APPROVED** [review]
What about creating an issue to handle the issue when the attendee accepts/declines the whole series again? We might discuss the behaviour in that issue too.

**Review by asmaNode — APPROVED** [review]

**Pull request merged**